exercises/http_uppercaserer/exercise.js

```diff
@@ -54,34 +54,43 @@ function query (mode) {
 
   function connect (port, stream) {
     var input = through2()
-    var count = 0
-    var iv
     var url = 'http://localhost:' + port
-    var req
-
-    // TODO: test GET requests for #fail
-    req = input.pipe(hyperquest.post(url)
-      .on('error', function (err) {
-        exercise.emit(
-          'fail'
-          , exercise.__('fail.connection', {address: url, message: err.message})
-        )
-      }))
-
-    req.pipe(stream)
-    setTimeout(function () {
-      stream.unpipe(req)
-      stream.end()
-    }, 5000)
-
-    iv = setInterval(function () {
-      input.write(words[count].trim() + '\n')
-
-      if (++count === words.length) {
-        clearInterval(iv)
-        input.end()
-      }
-    }, 50)
+    var methods = ['post', 'get']
+
+    var mi = 0
+    var miv = setInterval(function () {
```

Comment:
Can we get rid of the `setInterval` and `setTimeout` calls somehow? These make the code complex.